HW04.Rmd

```diff
@@ -1,126 +1,260 @@
+ggplot(monthly_death, aes(x = factor(month), y = deaths - 6000)) +
+    geom_bar(stat = "identity") +
+    scale_x_discrete("Month",
+                     breaks = 1:12,
+                     labels = c("Jan", "Feb", "Mar", "Apr", "May", "Jun", "Jul", "Aug", "Sep", "Oct", "Nov", "Dec")) + 
+    scale_y_continuous(name = "Deaths",
+                       limits = c(0,3200),
+                       breaks = c(0, 1000, 2000, 3000),
+                       labels = c("6000", "7000", "8000", "9000"),
+                       expand = c(0,0)) +
+    geom_text(aes(label = deaths), vjust = -1) +
+    labs(title = "Gun deaths per month") +
+    theme(panel.background = element_blank(),
+          axis.line = element_line(),
+          plot.title = element_text(hjust = 0.5))
+    
+```
+
+## Generating a bar chart that identifies the number of gun deaths associated with each type of intent cause of death. The bars should be sorted from highest to lowest values.
+
+```{r intent}
+intent_type <- gun_death %>%
+    count(intent, sort = TRUE) %>%
+    rename(deaths = n)
+ggplot(intent_type, aes(x = reorder(intent, -deaths), y = deaths)) +
+    geom_bar(stat = "identity") +
+    scale_y_continuous("Deaths",
+                       expand = c(0,0), 
+                       limits = c(0,70000)) +
+    labs(title = "Number of each type of intent") +
+    xlab("Intent") +
+    geom_text(aes(label = deaths), vjust = -1) +
+    theme(panel.background = element_blank(),
+          axis.line = element_line(),
+          plot.title = element_text(hjust = 0.5))
+```
+
+## Generating a boxplot visualizing the age of gun death victims, by sex. Print the average age of female gun death victims.
+
+```{r age_&_sex}
+age_sex <- gun_death %>%
+    select(age, sex)
+age_sex = subset(age_sex, age != "Not Available")
+age_sex$age<-as.numeric(age_sex$age)
+average_age <- age_sex %>%
+    group_by(sex) %>%
+    summarize(ave_age = mean(age))
+average_age$ave_age<-as.integer(average_age$ave_age)
+ggplot(age_sex, aes(x = sex, y = age, color = sex, fill = sex)) +
+    geom_boxplot(alpha = 0.4) +
+    annotate(geom = "text", x = 2, y = 50, label = paste("Average age:", average_age[1,2])) +
+    scale_x_discrete("Sex",
+                     labels = c("F" = "Female", "M" = "Male")) +
+    labs(title = "Age of gun death victims") +
+    ylab("Age") +
+    ##geom_text(aes(label = deaths), vjust = -1) +
+    theme(panel.background = element_blank(),
+          axis.line = element_line(),
+          plot.title = element_text(hjust = 0.5),
+          legend.position = "none")
+```
+
+
+## How many white males with at least a high school education were killed by guns in 2012?
+
+```{r education}
+white <- gun_death %>%
+    filter(sex == "M", race == "White", education != "Less than HS")
+```
+Using these filters, the new data frame "white" contains 47039 observations in total, which is the number of white males with at least a high school education level that were killed by guns in 2012.
```

Comment:
Hi Lai, 

I noticed a discrepancy in the values that we got. I got 15484 whereas you got a larger number, 47039. Upon closer inspection, I think youmay have missed adding in the filter for the year 2012 in your "white" data frame.